internal/logging/logging.go

```diff
@@ -0,0 +1,123 @@
+// Package logging implements the logger for the pack CLI.
+package logging
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/apex/log"
+	"github.com/fatih/color"
+
+	"github.com/buildpack/pack/style"
+)
+
+// Terminal colors
+const (
+	errorLevelText = "ERROR"
+	// time format the out logging uses
+	timeFmt = "2006/01/02 15:04:05.000000"
+)
+
+// handler implementation.
+type handler struct {
+	sync.Mutex
+	writer   io.Writer
+	wantTime bool
+	timer    func() time.Time
+}
+
+func formatLevel(ll log.Level) string {
+	if ll == log.ErrorLevel {
+		if color.NoColor {
+			return fmt.Sprintf("%-6s ", errorLevelText)
+		}
+		return style.Error("%-6s ", errorLevelText)
+	}
+	return ""
+}
+
+// preserve behavior of other loggers
+func appendMissingLineFeed( msg string ) string {
+	buff := []byte(msg)
+	if buff[len(buff)-1] != '\n' {
+		buff = append(buff, '\n')
+	}
+	return string(buff)
+}
+
+// HandleLog supports behavior that is unique to Pack CLI, namely toggling colors and timestamps.
+func (h *handler) HandleLog(e *log.Entry) error {
+	h.Lock()
+	defer h.Unlock()
+
+	// if we have a blank line we don't want padding or prefixes
+	if e.Message == "" {
+		_, _ = fmt.Fprintln(h.writer)
+		return nil
+	}
+
+	if h.wantTime {
+		ts := h.timer().Format(timeFmt)
+		_, _ = fmt.Fprint(h.writer, appendMissingLineFeed(fmt.Sprintf("%s %s%-25s", ts, formatLevel(e.Level), e.Message)))
+		return nil
+	}
+
+	_, _ = fmt.Fprint(h.writer, appendMissingLineFeed(fmt.Sprintf( "%s%-25s", formatLevel(e.Level), e.Message)))
+
+	return nil
+}
+
+type logWithWriters struct {
+	log.Logger
+	out io.Writer
+	errOut io.Writer
+	handler *handler
+}
+
+// Writer returns stdout if level is verbose (debug)
+func(lw *logWithWriters) Writer() io.Writer {
```

Comment:
Suggest calling this `DebugWriter` (similar to the old `VerboseWriter`)

Reply:
I'd disagree on this one. I think you're correct that `DebugWriter` is more clear for internal purposes however the tradeoff is that `Writer` is supported by external logging packages that a developer might want to plug into the pack module so `Writer` is part of the logging.Log contract.

Reply:
Ah, makes sense. I wonder if this method should always return the writer then, and we provide some other means (method, function) of getting one that's tied to the verbosity/debug setting.